### PR TITLE
Translation of Git episode 5

### LINKED ISSUE #65: Translation: git-novice Episode 5
Translation: git-novice Section 5 "Exploring History"
-----

* Please assign yourself (or add a comment) to the issue **before** you start translating, so other people know that you are translating.
* Once you have finished translating, send a pull request and reference this issue.

翻訳：git-novice セクション5 "Exploring History"
-----

* あなたが翻訳し始めたと分かるように、翻訳を**始める前**にこのイシューを自分に割り振って(assign)、あるいはコメントを書いて下さい。
* 翻訳が終わったら、プルリクエストを開いて、このイシューを参照・リンクして下さい。

https://github.com/swcarpentry-ja/i18n/blob/b3225373bbf9b2f14b8882e4bc019cbd9dbad114/po/git-novice.ja.po#L3736

### PR DESCRIPTION
**Pull request by kozo2**

### CODE REVIEW

**Comment by TomKellyGenetics** [comment]
お疲れ様でした。

File: po/git-novice.ja.po [hunk 1]
```diff
@@ -3712,18 +3712,18 @@ msgid ""
...
-"- \"How can I identify old versions of files?\"\n"
-"- \"How do I review my changes?\"\n"
-"- \"How can I recover old versions of files?\"\n"
-"objectives:\n"
-"- \"Explain what the HEAD of a repository is and how to use it.\"\n"
-"- \"Identify and use Git commit numbers.\"\n"
-"- \"Compare various versions of tracked files.\"\n"
-"- \"Restore old versions of files.\"\n"
+"title: 履歴の探索\n"
+"講義: 25\n"
```
Comment (TomKellyGenetics) [inline po/git-novice.ja.po]:
```suggestion
"teaching: 25\n"
```

File: po/git-novice.ja.po [hunk 2]
```diff
@@ -3712,18 +3712,18 @@ msgid ""
...
+"演習: 0\n"
```
Comment (TomKellyGenetics) [inline po/git-novice.ja.po]:
```suggestion
"exercises: 0\n"
```

File: po/git-novice.ja.po [hunk 3]
```diff
@@ -3712,18 +3712,18 @@ msgid ""
...
+"質問:\n"
```
Comment (TomKellyGenetics) [inline po/git-novice.ja.po]:
```suggestion
"questions:\n"
```

File: po/git-novice.ja.po [hunk 4]
```diff
@@ -3712,18 +3712,18 @@ msgid ""
...
+"- \"ファイルの古いバージョンを確認するにはどうすればよいでしょうか?\"\n"
+"- \"変更内容を再調査するにはどうすればよいでしょうか?\"\n"
+"- \"ファイルの古いバージョンを復元するにはどうすればよいでしょうか?\"\n"
+"目標:\n"
```
Comment (TomKellyGenetics) [inline po/git-novice.ja.po]:
```suggestion
"keypoints:\n"
```

**Review by TomKellyGenetics — COMMENTED** [review]
Note we don't need to change the headings.

File: po/git-novice.ja.po [hunk 5]
```diff
@@ -3712,18 +3712,18 @@ msgid ""
...
-"- \"How can I recover old versions of files?\"\n"
-"objectives:\n"
-"- \"Explain what the HEAD of a repository is and how to use it.\"\n"
-"- \"Identify and use Git commit numbers.\"\n"
-"- \"Compare various versions of tracked files.\"\n"
-"- \"Restore old versions of files.\"\n"
+"- \"ファイルの古いバージョンを確認するにはどうすればよいでしょうか?\"\n"
+"- \"変更内容を再調査するにはどうすればよいでしょうか?\"\n"
+"- \"ファイルの古いバージョンを復元するにはどうすればよいでしょうか?\"\n"
+"keypoints:\n"
```
Comment (TomKellyGenetics) [inline po/git-novice.ja.po]:
```suggestion
"objectives:\n"
```

**Comment by TomKellyGenetics** [comment]
@kozo2 is this one ready to check? It looks like this is a full lesson. 🎉

**Comment by kozo2** [comment]
@TomKellyGenetics Yes, I would like to ask you to check. Thank you.

File: po/git-novice.ja.po [hunk 6]
```diff
@@ -3712,21 +3712,21 @@ msgid ""
...
 "---\n"
-"title: Exploring History\n"
+"title: 履歴の探索\n"
 "teaching: 25\n"
 "exercises: 0\n"
 "questions:\n"
-"- \"How can I identify old versions of files?\"\n"
-"- \"How do I review my changes?\"\n"
-"- \"How can I recover old versions of files?\"\n"
+"- \"ファイルの古いバージョンを確認するにはどうすればよいでしょうか?\"\n"
```
Comment (joelnitta) [inline po/git-novice.ja.po]: The questions in the R lesson all end in ~ ですか？ I actually prefer ~でしょうか？, but I think we should be consistent one way or the other

File: po/git-novice.ja.po [hunk 7]
```diff
@@ -3951,16 +3944,13 @@ msgid ""
...
-"These are unique IDs for the changes,\n"
-"and \"unique\" really does mean unique:\n"
-"every change to any set of files on any computer\n"
-"has a unique 40-character identifier.\n"
-"Our first commit was given the ID\n"
-"`f22b25e3233b4645dabd0d81e651fe074bd8e73b`,\n"
-"so let's try this:"
+"`git log` が表示する、数字と文字の長い文字列を使用して\n"
+"コミットを参照することもできます。\n"
+"これらは変更に対するユニークなIDであり、「ユニーク」は本当に唯一であることを意味します:\n"
```
Comment (joelnitta) [inline po/git-novice.ja.po]: Not really needed to improve the translation, but rather to make the meaning clearer.
```suggestion
"これらは一個一個の変更に対するユニークなIDであり、「ユニーク」は本当に唯一であることを意味します:\n"
```

File: po/git-novice.ja.po [hunk 8]
```diff
@@ -3951,16 +3944,13 @@ msgid ""
...
+"任意のコンピューター上のファイルの任意のセットに対するすべての変更には、\n"
```
Comment (joelnitta) [inline po/git-novice.ja.po]:
```suggestion
"どのコンピューターのどのファイルの変更の組み合わせに対して、\n"
```

Comment (joelnitta) [reply po/git-novice.ja.po]: General question: why is it "on any computer"? 
If you made the exact same change to the exact same set of files on another computer, would it have a different SHA?

File: po/git-novice.ja.po [hunk 9]
```diff
@@ -4083,14 +4072,13 @@ msgid ""
...
-"which is the last saved commit.\n"
-"If we want to go back even further,\n"
-"we can use a commit identifier instead:"
+"その名前から推測できるように、\n"
+"`git checkout` はファイルの古いバージョンをチェックアウト (つまり、復元) します。\n"
+"この場合、\n"
+"最後に保存されたコミットである `HEAD` に記録されたファイルのバージョン "
+"を復元することをGitに伝えています。\n"
+"さらに戻りたい場合は、\n"
+"代わりにコミット識別子を使うことができます:"
```
Comment (joelnitta) [inline po/git-novice.ja.po]: I noticed in Lesson 7, @rikutakei uses 「コミット ID」。どっちかに統一した方がいいと思います。

File: po/git-novice.ja.po [hunk 10]
```diff
@@ -4210,13 +4194,13 @@ msgid ""
...
 "recent commit (`HEAD~1`), which is commit `f22b25e`:"
 msgstr ""
-"It's important to remember that\n"
-"we must use the commit number that identifies the state of the repository\n"
-"*before* the change we're trying to undo.\n"
-"A common mistake is to use the number of\n"
-"the commit in which we made the change we're trying to get rid of.\n"
-"In the example below, we want to retrieve the state from before the most\n"
-"recent commit (`HEAD~1`), which is commit `f22b25e`:"
+"元に戻す変更を行う *前* に、\n"
```
Comment (joelnitta) [inline po/git-novice.ja.po]: "before" refers to the order of the commits, not the order of commands in git
```suggestion
"取り消したい変更の一個**前**のコミット番号を使う\n"
```

File: po/git-novice.ja.po [hunk 11]
```diff
@@ -4210,13 +4194,13 @@ msgid ""
...
+"リポジトリの状態を識別するコミット番号を使用する必要がある\n"
```
Comment (joelnitta) [inline po/git-novice.ja.po]:
```suggestion
"必要がある\n"
```

File: po/git-novice.ja.po [hunk 12]
```diff
@@ -4294,19 +4276,19 @@ msgid ""
...
-"The fact that files can be reverted one by one\n"
-"tends to change the way people organize their work.\n"
-"If everything is in one large document,\n"
-"it's hard (but not impossible) to undo changes to the introduction\n"
-"without also undoing changes made later to the conclusion.\n"
-"If the introduction and conclusion are stored in separate files,\n"
-"on the other hand,\n"
-"moving backward and forward in time becomes much easier."
+"ファイルを1つずつ元に戻すことができるという事実は\n"
+"人々が研究を整理する方法を変えるのに役立ちます。\n"
```
Comment (joelnitta) [inline po/git-novice.ja.po]: It's not saying "this helps people change their work style", just that it often has this effect.
```suggestion
"人々が研究を整理する方法を変えることがあります。\n"
```

File: po/git-novice.ja.po [hunk 13]
```diff
@@ -4384,31 +4365,31 @@ msgid ""
...
-"> 2. Copy the ID (the first few characters of the ID, e.g. 0b1d055).\n"
+"> 2. そのIDをコピーします (IDの最初の数文字は例えば 0b1d055)。\n"
 ">\n"
 "> 3. `git revert [commit ID]`\n"
 ">\n"
-"> 4. Type in the new commit message.\n"
+"> 4. 新しいコミットメッセージを入力します。\n"
 ">\n"
-"> 5. Save and close"
+"> 5. 保存して閉じます"
```
Comment (joelnitta) [inline po/git-novice.ja.po:4388]: This is odd.. it's not a problem with the translation, but I can't find the answer to the quiz question here (or the one before it).

File: po/git-novice.ja.po [hunk 14]
```diff
@@ -4541,20 +4522,19 @@ msgid ""
...
+"このコマンドを実行したら\n"
+"> どうなるだろうと予測しますか? 実行すると何が起こっていますか? またそれはなぜでしょうか?\n"
 ">\n"
-"> Try another command, `git diff [ID] mars.txt`, where [ID] is replaced "
-"with\n"
-"> the unique identifier for your most recent commit. What do you think will "
-"happen,\n"
-"> and what does happen?"
+"> 別のコマンド `git diff [ID] mars.txt` を試しましょう、ここでの [ID] は "
+"> 最新のコミットのユニークな識別子で置き換えられます。 "
```
Comment (joelnitta) [inline po/git-novice.ja.po]: Should be consistent about ユニークな識別子 vs 識別子 vs コミット ID, etc

File: po/git-novice.ja.po [hunk 15]
```diff
@@ -4568,18 +4548,17 @@ msgid ""
...
 msgstr ""
 ">\n"
-"> `git checkout` can be used to restore a previous commit when unstaged "
-"changes have\n"
-"> been made, but will it also work for changes that have been staged but not "
-"committed?\n"
-"> Make a change to `mars.txt`, add that change, and use `git checkout` to "
-"see if\n"
-"> you can remove your change."
+"> `git checkout` は、ステージされていない変更が行われたときに "
```
Comment (joelnitta) [inline po/git-novice.ja.po]:
```suggestion
"> `git checkout` は、ステージされていない変更があったときに "
```

File: po/git-novice.ja.po [hunk 16]
```diff
@@ -4568,18 +4548,17 @@ msgid ""
...
+"以前のコミットを復元するために使用できます。\n"
+"> しかしそれはステージされているがコミットされていない変更に対しても機能する"
+"でしょうか?\n"
+"> `mars.txt` に変更を用意し、その変更を加え、`git checkout` を使い "
```
Comment (joelnitta) [inline po/git-novice.ja.po]: Not in the original, but I think it's needed for clarity
```suggestion
"> `mars.txt` に変更を用意し、その変更を加え（`git add`を使い）、`git checkout` を使い "
```

File: po/git-novice.ja.po [hunk 17]
```diff
@@ -4626,41 +4605,41 @@ msgid ""
 "> {: .language-bash}"
 msgstr ""
 ">\n"
-"> Exploring history is an important part of git, often it is a challenge to "
-"find\n"
-"> the right commit ID, especially if the commit is from several months ago.\n"
+"> 履歴の探索はgitの重要な要素であり、特にそのコミットが数か月前のものである場合は、 "
```
Comment (joelnitta) [inline po/git-novice.ja.po]:
```suggestion
"> 履歴の探索はgitの重要な要素であり、特にそのコミットが数ヶ月前のものである場合は、 "
```

File: po/git-novice.ja.po [hunk 18]
```diff
@@ -4626,41 +4605,41 @@ msgid ""
...
-"`update files`.\n"
-"> How can you search through these files?\n"
+"> 残念ながら、これらのコミットメッセージの一部は非常にあいまいです。例えば、 "
+"`update files`。\n"
+"> どうやったらこれらのファイルを念入りに調べることができますか?\n"
 ">\n"
-"> Both `git diff` and `git log` are very useful and they summarize a "
-"different part of the history for you.\n"
-"> Is it possible to combine both? Let's try the following:\n"
+"はどちらも非常に便利で、履歴のさまざまな部分を要約しています。"
```
Comment (joelnitta) [inline po/git-novice.ja.po]: This line appears to be included by mistake. すぐ下の行と同じような内容ですが、間違いなのかな？この行を完全に取り消した方がいいと思われます。
```suggestion
""
```

**Review by joelnitta — COMMENTED** [review]
Thanks for the PR!
Please see my comments and suggestions.

File: po/git-novice.ja.po (hunk 6 above)
Comment (kozo2) [reply po/git-novice.ja.po]: Replaced でしょうか to ですか https://github.com/kozo2/i18n/commit/2fce05d5947cc1ad8e6b98afde1fecaed04d8f5f

File: po/git-novice.ja.po (hunk 9 above)
Comment (kozo2) [reply po/git-novice.ja.po]: https://github.com/kozo2/i18n/commit/5c038cd3864fa34c1bd9bca5f56ad2dbc1ba1fb7

**Review by TomKellyGenetics — APPROVED** [review]

File: po/git-novice.ja.po (hunk 14 above)
Comment (kozo2) [reply po/git-novice.ja.po]: Replaced 識別子 to ID
https://github.com/kozo2/i18n/commit/c184e2362633755a96c0336c20ee3a3d70193fe4

File: po/git-novice.ja.po (hunk 18 above)
Comment (kozo2) [reply po/git-novice.ja.po]: 間違いでした。不要な内容を消し忘れていました。

File: po/git-novice.ja.po (hunk 13 above)
Comment (kozo2) [reply po/git-novice.ja.po:4388]: Good point.
I also can't find the "Solution" (== the answer)...
https://swcarpentry.github.io/git-novice/05-history/index.html#solution-1
What should we do...

File: po/git-novice.ja.po (hunk 8 above)
Comment (kozo2) [reply po/git-novice.ja.po]: I think this means that every computer has an ID with the same "number of characters" of "40 characters", and does not mean that the "contents" of "40 characters" are the same.
According to this blog entry (https://engineering.mercari.com/blog/entry/2016-02-08-173000/),
if the author and committer information is different, it seems to have a different SHA.
(And whether the changes are the same or different does not seem to affect the SHA.)

**Comment by kozo2** [comment]
I resolved all review comments except that there is no original English text for "Solution".

**Comment by joelnitta** [comment]
Thanks! Let's merge this and file that as an issue to fix later then.

**Review by joelnitta — APPROVED** [review]

**Comment by TomKellyGenetics** [comment]
@joelnitta @kozo2 
I've looked into the missing "solutions" issue. I've checked the [original lesson](https://swcarpentry.github.io/git-novice/05-history/index.html#solution) and could not find that text in the [git-novice.ja.po](https://github.com/swcarpentry-ja/i18n/blob/ja/po/git-novice.ja.po). It seems these solutions were not included in [our English version](https://swcarpentry-ja.github.io/git-novice/05-history/index.html) either. It's unclear to me whether our English version is simply outdated or the hidden [solution](https://github.com/swcarpentry/git-novice/blob/gh-pages/_episodes/05-history.md#solution) feature isn't supported in our theme/po4gitbook workflow.

Anyway well done on getting the lesson translated.

**Pull request merged**